aiohttp/web_request.py

```diff
@@ -664,6 +665,18 @@ def app(self):
         """Application instance."""
         return self._match_info.current_app
 
+    @property
+    def config_dict(self):
+        lst = self._match_info.apps
+        app = self.app
+        idx = lst.index(app)
+        sublist = list(reversed(lst[:idx + 1]))
+        return ChainedProps(sublist)
+
+    @property
+    def config(self):
+        return Namespace(self.config_dict)
```

Comment:
I'm in a very doubt if we need not a mapping class here. This will cause two problems:
1. Inconsistent access to config values. While names are valid Python attributes it's all fine, but how to access to a value with `mycomponent.cache`? Here, user have to fallback to `getattr(request.config, 'mycomponent.cache')`.
2. Linters. Linters will complain about accessing to unknown class attributes. This will be very annoying. 

I would prefer to have one way to work with the config that works and right. So far, `config_dict` is more powerful and useful interface to it.

Reply:
I heard complains many times that IDE doesn't autocomplete string keys but does it for custom attributes.

Reply:
Yes, they don't do keys autocomplete unless you explicitly define those keys in the same scope. However, their ability to autocomplete custom attributes is limited and based on type knowledge and similar usages. 

However, for pylint you will have to put Namespace class into  `ignored-classes` since it will think that it's your fault since you're referencing to unknown class instance attribute.

Reply:
@socketpair I recall you wanted the feature.
Please explain your opinion

Reply:
@Deepwalker ?